.github/workflows/ci-docs.yml

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Docs
+
+on:
+  pull_request:
+  push:
+
+env:
+  STRUCTOR_VERSION: v1.7.1
+  BUILD_MAVEN_OPTS: "-Dmaven.javadoc.skip=true -DskipTests=true --batch-mode --also-make"
+  VERIFY_MAVEN_OPTS: "-Pcoverage"
+
+jobs:
+  build-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install --projects janusgraph-all ${{ env.BUILD_MAVEN_OPTS }}
+
+  build-doc:
+    runs-on: ubuntu-20.04
+    needs: build-all
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install -pl janusgraph-doc ${{ env.BUILD_MAVEN_OPTS }}
+      - run: git diff  --exit-code docs/basics/janusgraph-cfg.md
+      - run: docker build -t doc-site:mkdocs -f docs/docs.Dockerfile ./docs
+      - run: docker run --rm -v $PWD:/mkdocs doc-site:mkdocs mkdocs build
+      - uses: actions/upload-artifact@v2
+        if: github.ref != 'ref/head/master'
```

Comment:
These are not uploaded on master because we deploy them there completely?

Reply:
We deploy the documentation on the master. I think their is no reason store the artefacts.